Mark Trustar as FIPS compliant

shiyingtu opened the pull request:
- Mark as fips_compliant
- Fix linting tests
- Fix semgrep tests
- Add verify argument

shiyingtu committed:
Mark Trustar as FIPS compliant

shiyingtu commented:
Static failing tests:

- No Code Consts: Line 51 has value [-1,0,1,2,3]. -1 does not satisfy the ast types in test code.

- Min Platform Version

Integration failing test:

- This is a known issue. Track and will be fixed in PAPP-23064

shiyingtu committed:
Fix linting tests

shiyingtu committed:
Fix somgrep tests

paulnawat commented on trustar_connector.py:
```diff
@@ -227,8 +230,10 @@ def _make_rest_call_helper(self, endpoint, action_result, headers={}, params=Non
             # If token is expired, generate a new token
-            if error_message and (consts.TRUSTAR_INVALID_TOKEN_MESSAGES[0] in error_message or consts.TRUSTAR_INVALID_TOKEN_MESSAGES[1] in error_message):
-                self.debug_print("Refreshing TRUSTAR API and re-trying request to [{}] because API token was expired or invalid with error [{}]".format(endpoint, error_message))
+            if error_message and (consts.TRUSTAR_INVALID_TOKEN_MESSAGES[0]
+                    in error_message or consts.TRUSTAR_INVALID_TOKEN_MESSAGES[1] in error_message):
```
Should we use brackets inside the 2nd part?
```
((consts.TRUSTAR_INVALID_TOKEN_MESSAGES[0] in error_message) or (consts.TRUSTAR_INVALID_TOKEN_MESSAGES[1] in error_message))
```
It might be easier to indent and easier to read later on.

shiyingtu replied:
Done

paulnawat commented on trustar_connector.py:
```diff
@@ -195,6 +194,10 @@ def _make_rest_call_helper(self, endpoint, action_result, headers={}, params=Non
 
         retry_failure_flag = False
+
+        if headers is None:
+            headers = {}
```
What do you think about:
```
if not headers:
  headers = {}
```
here we can also turn other False value into an empty object

shiyingtu replied:
Done

paulnawat reviewed (dismissed):
Changes look great. I left 2 minor suggestion / question.

shiyingtu committed:
Update files suggested by pre-commit and refactor some code

shiyingtu committed:
Impose stricter rules on headers as an empty object

shiyingtu commented:
Note for Release: static test failures tracked in PAPP-23197

shiyingtu committed:
Enable Certificate verification for request calls

shiyingtu committed:
Add local tox.ini file

shiyingtu committed:
Refactor verify argument for request calls

- Background: semgrep tests require request calls to be
certificate verified.
- Fix: Without changing the previous behaviour, add a verify
argument so that users can customize the option. The default is
set to false.
